core/templates/dev/head/domain/exploration/StateTopAnswerStatisticsFactory.js

```diff
@@ -0,0 +1,120 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for domain object which holds the list of top answer
+ * statistics for a particular state.
+ */
+
+oppia.factory('StateTopAnswerStatisticsFactory', [
+  '$http', '$injector', 'AngularNameService', 'AnswerClassificationService',
+  'ExplorationContextService', 'ExplorationStatesService',
+  function(
+      $http, $injector, AngularNameService, AnswerClassificationService,
+      ExplorationContextService, ExplorationStatesService) {
+    /**
+     * @typedef {Object} AnswerStatistics - A record for the statistics of a
+     * single top answer.
+     *
+     * @property {*} answer - Contains the answer in its raw form, directly from
+     *    the backend.
+     * @property {string} answerAsHtml - Contains the answer as a string which
+     *    can be rendered directly as HTML.
+     * @property {number} frequency
+     * @property {boolean} isAddressed
+     */
+
+    /**
+     * TODO(brianrodri): Move this helper function into a proper service which
+     * takes the state's interaction type into account when formatting the HTML.
+     * @param {*} answer
+     * @returns {string}
+     */
+    var convertAnswerToHtml = function(answer) {
+      // Don't render quotes when the answer is a string.
+      return (typeof answer === 'string') ? answer : JSON.stringify(answer);
+    };
+
+    /**
+     * @private @constructor
+     * Returns a state's top answers with stale data. For a fresh instance, use
+     * {@link StateTopAnswerStatistics#createFromBackendDict} instead, or call
+     * {@link StateTopAnswerStatistics#refreshIsAddressed} after construction.
+     *
+     * @param {!string} stateName
+     * @param {!{answer, frequency: number}[]} backendTopAnswers
+     */
+    var StateTopAnswerStatistics = function(stateName, backendTopAnswers) {
+      /** @type {string} */
+      this._stateName = stateName;
+      /** @type {AnswerStatistics[]} */
+      this._answers = backendTopAnswers.map(function(answerFrequencyPair) {
+        return {
+          answer: angular.copy(answerFrequencyPair.answer),
+          answerAsHtml: convertAnswerToHtml(answerFrequencyPair.answer),
+          frequency: answerFrequencyPair.frequency,
+          isAddressed: false, // stale, needs to be updated.
+        };
+      });
+    };
+
+    /** Examines associated state to refresh the addressed info of answers. */
+    StateTopAnswerStatistics.prototype.refreshIsAddressed = function() {
```

Comment:
It's a bit weird for a domain object class to depend on so many other services. You might consider moving this functionality out into the relevant service instead in the future (maybe add a TODO).

Reply:
Hmm, I was thinking of going along w/ the design of StatesObjectFactory and StateObjectFactory, but I kept the singular form around as a simple typedef since I didn't want to give it too much data.

I went along w/ this b/c I found it easier to keep initialization separate since it takes the most dependencies. Keeping them updated really only requires a bunch of calls to refreshIsAddressed (which I'll be renaming).